lib/modules/actions.js

```diff
@@ -54,7 +56,7 @@ export const reroutePage = () => async (dispatch, getState) => {
       queryParams: currentPage.queryParams,
       hashParams: currentPage.hashParams,
       bodyParams: {},
-      referrer: currentPage.referrer,
+      referrer: currentPage.referrer || NO_REFERRER,
```

Comment:
For clarification - the app does not initialize twice, what is duplicated is the first page in history. If you refresh the page, in redux dev tools you will see this

![screen shot 2016-12-21 at 10 55 43 am](https://cloud.githubusercontent.com/assets/7064600/21402170/211ab2a0-c76c-11e6-8f6b-4d8155a023e3.png)